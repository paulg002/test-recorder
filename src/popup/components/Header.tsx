import * as React from 'react';
import InfoButton from './InfoButton';

export interface HeaderProps {
  toggleInfoDisplay: () => void,
  shouldInfoDisplay: boolean,
}

export default ({ shouldInfoDisplay, toggleInfoDisplay }: HeaderProps) => (
  <div id="header">
    <h1 id="title">Test recorder</h1>
    <InfoButton shouldInfoDisplay={shouldInfoDisplay} toggleInfoDisplay={toggleInfoDisplay} />
  </div>
);
