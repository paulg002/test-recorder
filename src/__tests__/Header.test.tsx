import * as React from 'react';
import { shallow } from 'enzyme';
import Header, { HeaderProps } from '../popup/components/Header';
import InfoButton from '../popup/components/InfoButton';
import '../setupTests';


describe('Header', () => {
  let wrapper;
  let props: HeaderProps;
  beforeAll(() => {
    props = {
      shouldInfoDisplay: false,
      toggleInfoDisplay: jest.fn(),
    };
  });
  it('Should render Header with title of Test recorder', () => {
    wrapper = shallow(<Header { ...props } />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.exists(InfoButton)).toBe(true);
    const title = wrapper.find('#title');
    expect(title.text()).toBe('Test recorder');
  });
});
