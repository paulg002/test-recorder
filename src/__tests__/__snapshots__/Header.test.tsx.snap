// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should render Header with title of Test recorder 1`] = `
<div
  id="header"
>
  <h1
    id="title"
  >
    Test recorder
  </h1>
  <Component
    shouldInfoDisplay={false}
    toggleInfoDisplay={[MockFunction]}
  />
</div>
`;
